### DISCUSSION TITLE
AnyAddress(pubKey,CoinType.TRON) in Android is wrong

### DISCUSSION THREAD

**Discussion by Daley** [trustwallet/wallet-core#3486] [Q&A]
**Describe the bug**
here is my code ,i want use AnyAddress to get address from PublicKey
`var key=hdWallet.getKey(CoinType.TRON,"m/44'/195'/0'/0/1")`
`var pub=key.getPublicKeySecp256k1(false)`
 `var anyAddr= AnyAddress(pub,CoinType.TRON)`

but !! 
anyAddr.data().size() is 0 

**Expected behavior**
I want transfer anyAddr.data() to hex string , so i can get address

**Additional context**
Maybe there is another to get Address from PublickKey ,pls tell me.

**Comment by satoshiotomakan** [comment]
Hi @Daley, could you please confirm if the given `pub` has an expected value?

**Reply by Daley** [reply]
i print `pub` with `toHexString(pub.data())`  i get same string with TronWeb.js

**Reply by Daley** [reply]
![image](https://github.com/trustwallet/wallet-core/assets/249557/71839736-a430-416e-bc83-c5af85a7b4e1)
